src/plugins/imagery/components/imageryViewLayout.vue

```diff
@@ -0,0 +1,294 @@
+            }
+
+            return false;
+        },
+        getImageUrl(datum) {
+            return datum ?
+                this.imageFormat.format(datum) :
+                this.imageUrl;
+        },
+        getTime(datum) {
+            return datum ?
+                this.timeFormat.format(datum) :
+                this.time;
+        },
+        handleScroll() {
+            const thumbsWrapper = this.$refs.thumbsWrapper
+            if (!thumbsWrapper) {
+                return;
+            }
+
+            const { scrollLeft, scrollWidth, clientWidth, scrollTop, scrollHeight, clientHeight } = thumbsWrapper;
+            const disableScroll = (scrollWidth - scrollLeft) > 2 * clientWidth
+                    || (scrollHeight - scrollTop) > 2 * clientHeight;
+            this.autoScroll = !disableScroll;
+        },
+        paused(state) {
+            if (arguments.length > 0 && state !== this.isPaused) {
+                this.unselectAllImages();
+                this.isPaused = state;
+                if (this.nextDatum) {
+                    this.updateValues(this.nextDatum);
+                    delete this.nextDatum;
+                } else {
+                    this.updateValues(this.imageHistory[this.imageHistory.length - 1]);
+                }
+
+                this.autoScroll = true;
+            }
+
+            return this.isPaused;
+        },
+        requestHistory(bounds) {
+            this.requestCount++;
+            this.imageHistory = [];
+            const requestId = this.requestCount;
+            this.openmct.telemetry
+                .request(this.domainObject, bounds)
+                .then(function (values = []) {
+                    if (this.requestCount > requestId) {
+                        return Promise.resolve('Stale request');
+                    }
+
+                    values.forEach(function (datum) {
+                        this.updateHistory(datum);
+                    }, this);
+
+                    this.updateValues(values[values.length - 1]);
+                }.bind(this));
+        },
+        scrollToBottom() {
+            if (this.isPaused || !this.$refs.thumbsWrapper || !this.autoScroll) {
+                return;
+            }
+
+            const scrollHeight = this.$refs.thumbsWrapper.scrollHeight || 0;
+            if (!scrollHeight) {
+                return;
+            }
+
+            setTimeout(() => this.$refs.thumbsWrapper.scrollTop = scrollHeight, 0);
+        },
+        setSelectedImage(image) {
+            this.imageUrl = this.getImageUrl(image);
+            this.time = this.getTime(image);
+            this.paused(true);
+            this.unselectAllImages();
+            image.selected = true;
+        },
+        getElSize() {
+            if (!this.$el) {
+                return;
+            }
+
+            this.elementHeight = this.$el.parentElement.offsetHeight;
+            return this.elementHeight;
```

Comment:
Since the method is returning elementHeight, why not name it `getElementHeight`? Normally a getter does not set a value of a property, it only returns the value. It's confusing that it's doing both. How about defining `getElementHeight()` and `setElementHeight()`? You can call `setElementHeight()` in `mounted` and use it for the `resize` click handler also. The initial value of elementHeight in `data` can be set via `getElementHeight()`. Or just omit the getter and inline the value since it's not needed anywhere else.

Reply:
same as above, old code removed.